### PR TITLE
Fix build on riscv64gc-unknown-linux-gnu

### LINKED ISSUE #36: jemalloc-sys: bundled configure fails for riscv64gc
Hi,

jemalloc-sys currently fails to build from source on Debian (see [buildd logs](https://buildd.debian.org/status/logs.php?pkg=rust-jemalloc-sys)). The cause is that `configure` didn't recognize riscv64gc.

Testing reveals that, when using its `autogen.sh`, jemalloc successfully builds itself. However, with the current setting of this crate's `build.rs`, a bundled, pre-generated `configure` is used, which fails.

Running `autogen.sh` seems trivial, and it's available since jemalloc is submoduled. Is there anything that prevents its use?

### PR DESCRIPTION
**Pull request by Amanieu**
This works around the mismatch between the GNU target and rustc target.

Fixes #36

### CODE REVIEW

**Review by BusyJay — APPROVED** [review]

**Comment by BusyJay** [comment]
It seems there still be issues https://github.com/jemalloc/jemalloc/issues/2323.

/cc @bnoctis can you verify whether this works for you?

**Comment by bnoctis** [comment]
@BusyJay AFAICT this masks itself as riscv64, which is what vanilla jemalloc currently does. I don't know if there's any difference between it and riscv64gc. Upstream has a separate riscv64be, I think it's better to have a separate gc too.

**Comment by Amanieu** [comment]
The GNU toolchain prefix for RV64GC is riscv64-linux-gnu-. This is what the configure script is looking for. If you pass in riscv64gc- then it will fail to find the GCC binary.

**Comment by Amanieu** [comment]
At the very least this PR fixes the build for me when cross compiling from x86_64 to RV64.

File: jemalloc-sys/build.rs [hunk 1]
```diff
@@ -188,8 +188,9 @@ fn main() {
     }
 
     if let Ok(malloc_conf_opts) = read_and_watch_env("JEMALLOC_SYS_WITH_MALLOC_CONF") {
-        malloc_conf += &format!(
-            "{}{}",
+        malloc_conf = format!(
```
Comment (BusyJay) [inline jemalloc-sys/build.rs:191]: I think the expected fix for the warning should be
```
malloc_conf.push_str(xxx);
```

Comment (Amanieu) [reply jemalloc-sys/build.rs]: Done.

File: jemalloc-sys/build.rs [hunk 2]
```diff
@@ -188,11 +188,11 @@ fn main() {
     }
 
     if let Ok(malloc_conf_opts) = read_and_watch_env("JEMALLOC_SYS_WITH_MALLOC_CONF") {
-        malloc_conf += &format!(
+        malloc_conf.push_str(&format!(
```
Comment (BusyJay) [inline jemalloc-sys/build.rs:191]: Oh, what I mean is actually:
```
if !malloc_conf.is_empty() {
    malloc_conf.push(',');
}
malloc_conf.push_str(&malloc_conf_opts);
```

Comment (Amanieu) [reply jemalloc-sys/build.rs]: Done

**Comment by Amanieu** [comment]
I'm not sure why CI is failing, it works fine on my machine. Can we try a re-run?

**Comment by BusyJay** [comment]
Thank you @Amanieu @bnoctis

**Comment by Amanieu** [comment]
Could you publish a new version with this fix? Thanks!

**Comment by BusyJay** [comment]
Published as 0.5.2

**Pull request merged**